### DISCUSSION TITLE
Segmented progress bar

### DISCUSSION THREAD

**Discussion by Orholt** [AvaloniaUI/Avalonia#6333] [Q&A]
Hi, is there any known way to make segments on progress bar? 

Also i will like to know hot to assign specycif colors to segments for ex.
I have a progress bar with values from `0` to `100`
And i want to split it to 4 parts `25` each, also i want to make it visible to user for ex make border surrounding each segment

In range `0 - 25` i want to make progress bar red
In range `25 - 75` i want to make it yellow and
In range `75 - 100` i want it green

Thanks for any answers

**Comment by renrutsirhc** [comment]
What about stacking 3 progressbars on top of each other, one red, one yellow, one green. Give them a range of 25, 50, 25 and only set the yellow one's value once the red is full and the green once the yellow is full?

**Reply by Orholt** [reply]
There should be easier way, I guess achieved by styling

**Comment by maxkatz6** [comment]
> There should be easier way, I guess achieved by styling

Yes. If you need segments always visible, use VisualBrush as a background or just retemplate your progress bar using template from the source code as a reference.
If you need color changing depending on the value, you can use converter from decimal to the brush.

**Reply by Orholt** [reply]
Hi, could you give me an example code?
